library/src/main/java/me/mattlogan/library/ViewStack.java

```diff
@@ -3,58 +3,82 @@
 import android.animation.Animator;
 import android.animation.AnimatorListenerAdapter;
 import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
 import java.util.ArrayList;
 import java.util.EmptyStackException;
 import java.util.List;
-import java.util.Stack;
 
 import static me.mattlogan.library.Preconditions.checkNotNull;
 import static me.mattlogan.library.Preconditions.checkStringNotEmpty;
 
 /**
- * This manages a navigation stack by representing each item in the stack as a ViewFactory, which is
+ * This manages a navigation stack by representing each item in the stack as a layout id, which is
  * responsible for View creation. All standard Java Stack operations are supported, with additional
  * methods for pushing and popping with animated transitions.
  */
 public final class ViewStack {
 
-    private final Stack<ViewFactory> stack = new Stack<>();
+    public static final int SINGLE_VIEW = -1;
+    private final ParcelableIntStack stack = new ParcelableIntStack();
     private final ViewGroup container;
     private final ViewStackDelegate delegate;
     private final List<StackChangedListener> listeners = new ArrayList<>();
+    private final PancakesViewInflator pancakesViewInflator;
 
     /**
      * Static creation method for ViewStack instances
      *
-     * @param container Any ViewGroup container for navigation Views. Typically a FrameLayout
-     * @param delegate  A ViewStackDelegate responsible for "finishing" the navigation stack
+     * @param container Any {@link ViewGroup} container for navigation Views. Typically a FrameLayout
+     * @param delegate  A {@link ViewStackDelegate} responsible for "finishing" the navigation stack
      * @return A new ViewStack instance
      */
     public static ViewStack create(ViewGroup container, ViewStackDelegate delegate) {
+        return create(container, delegate, new PancakesViewInflator() {
+            @Override
+            public View inflateView(@LayoutRes int layoutResource, ViewGroup container) {
+                return LayoutInflater.from(container.getContext()).inflate(layoutResource, container, false);
+            }
+        });
+    }
+
+    /**
+     * Package-private factory method which allows us to provide a custom
+     * {@link PancakesViewInflator} for testing.
+     *
+     * @param container Any {@link ViewGroup} container for navigation Views.  Typically a FrameLayout
+     * @param delegate A {@link ViewStackDelegate} responsible for "finishing" the navigation stack
+     * @param pancakesViewInflator A {@link PancakesViewInflator} which is responsible for inflating
+     *                             layout resources into the container.
+     *
+     * @return A new ViewStack instance
+     */
+    static ViewStack create(ViewGroup container, ViewStackDelegate delegate, PancakesViewInflator pancakesViewInflator) {
```

Comment:
This is a cool pattern! If we want to keep this as a JVM test, I think this is the best option.

I'm actually leaning towards converting this whole thing to an instrumentation test that takes a real ViewGroup argument. Mocking concrete classes is generally a bad practice, especially super-complicated Android classes like ViewGroup. The API is simple enough, but I think we can't really call it a test unless we're actually using ViewGroup's implementation. That would also take away the need for all this ceremony around setting up our test objects.

Furthermore, if we convert this to an instrumentation test, we might be able to prevent future bugs caused by weirdness in the Android SDK like we saw with the Stack being cast to an ArrayList.

We can also add another instrumentation test later to address https://github.com/mattlogan/Pancakes/issues/10 (probably in a separate PR).

What do you think about making this an instrumentation test?